### PR TITLE
(PIE-455) Show the deprecation warning only when the report parameter

### PR DESCRIPTION
**Pull request by gregohardy**
has actually been defined.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -56,9 +56,15 @@
       minute  => '*/2',
     }
   }
-
+  $report_set = false
   if $enable_reports {
-    if $reports != '' {
+    if $reports == unset {
```
Comment (RandomNoun7) [inline manifests/init.pp:61]: Is this supposed to be `undef`?

Comment (gregohardy) [reply manifests/init.pp]: So apparently you cant set a string param to undef unless it's optional?

Comment (gregohardy) [reply manifests/init.pp]: It was a typo though, you are correct.

**Review by RandomNoun7 — APPROVED** [review]
I like it. I don't think the switch to Optional is a breaking change at all.

**Pull request merged**